orbstation/antagonists/traitor/objectives/final_objective/malware_injection.dm

```diff
@@ -0,0 +1,417 @@
+	target.update_appearance()
+
+// MALWARE INJECTION CONSOLE
+
+// 15 * 8 = 2 minutes total
+#define TIME_PER_STAGE 15 SECONDS
+#define CORRUPTION_STAGES 8
+
+/obj/machinery/computer/malware
+	name = "malware injection console"
+	desc = "A highly illegal console used by the Syndicate to inject an AI with malware."
+	icon_screen = "commsyndie"
+	icon_keyboard = "syndie_key"
+	circuit = /obj/item/circuitboard/computer/malware
+	///AI to inject malware into.
+	var/mob/living/silicon/ai/current = null
+	///Set to TRUE when transfering the virus into the console with a malware injector, and while the AI is being corrupted.
+	var/injecting = FALSE
+	///Set to TRUE when trying to resume the malware injection after it gets interrupted for whatever reason.
+	var/resuming = FALSE
+	///Current stage of corruption. The AI will become a malf AI when this number reaches CORRUPTION_STAGES.
+	var/corruption_stage = 0
+	///List of hacked APCs. This data is transferred to this computer through the malware injector. Used to retroactively connect the malf AI to the hacked APCs.
+	var/list/hacked_apcs = list()
+
+/obj/machinery/computer/malware/interact(mob/living/user)
+	if(!istype(user))
+		return
+	if(!user.mind.has_antag_datum(/datum/antagonist/traitor))
+		to_chat(user, span_alert("Access denied."))
+		return
+	if(injecting)
+		to_chat(user, span_alert("ERROR: Malware injection in progress."))
+		return
+	if(!injecting && corruption_stage && !resuming) // the injection had been aborted at some point
+		user.visible_message(span_danger("[user] hunches over [src] and begins typing away!"), span_notice("You start typing on [src] to resume the malware injection."))
+		resuming = TRUE
+		if(!do_after(user, INJECTOR_HACK_TIME, src, IGNORE_SLOWDOWNS, extra_checks = CALLBACK(src, .proc/inject_check)))
+			resuming = FALSE
+			return
+		resuming = FALSE
+		to_chat(user, span_boldnotice("Resuming malware injection."))
+		// time to go loud!
+		trigger_alarm(user) // set off the alarm, close and lock all doors in the area, alert the AI in big bold letters
+		announce_corruption()
+		advance_corruption_stage(user)
+		return
+	current = select_active_ai(user) // z-level does not matter for this
+	if(!current)
+		to_chat(user, span_alert("No active AIs detected!"))
+	else
+		to_chat(user, span_notice("[current.name] selected for malware injection."))
+
+/obj/machinery/computer/malware/attackby(obj/item/O, mob/living/user, params)
+	if(istype(O, /obj/item/malware_injector))
+		var/obj/item/malware_injector/M = O
+		if(machine_stat & (NOPOWER|BROKEN|MAINT))
+			return
+		if(!istype(user))
+			return
+		if(!user.mind.has_antag_datum(/datum/antagonist/traitor))
+			to_chat(user, span_warning("You have no idea how to use [M]."))
+			return
+		if(M.spent)
+			to_chat(user, span_alert("[M] is no longer functional."))
+			return
+		if(M.hacking)
+			to_chat(user, span_alert("[M] is already hacking something else."))
+			return
+		if(length(M.hacked_apcs) < APCS_REQUIRED_FOR_MALF)
```

Comment:
As you just use this define every time and the value is never modified you don't need it in the datum or in the hack tool, I think.